lib/js/asyncjs.nim

```diff
@@ -0,0 +1,60 @@
+import jsffi
+import macros
+
+when not defined(js):
+  {.error "asyncjs is only available for javascript"}
+
+
+type
+  Future*[T] = ref object
+    future*: T
+
+  PromiseJs* {.importcpp: "Promise".} = ref object
+
+proc generateJsasync(arg: NimNode): NimNode
```

Comment:
Reorder to avoid this foward decl.

Reply:
Done, removed all forward decl-s (and one unneeded `$` decl)